src/ws/Socket.ts

```diff
@@ -461,16 +461,16 @@ export class Socket extends EventEmitter {
 
         // If we already authed, it means we're reconnecting and should
         // establish authentication again.
-        if (this._authpacket) {
+        if (this._authpacket || this._optOutEventsArgs.length > 0) {
             // tslint:disable-next-line no-floating-promises
-            this.call(authMethod, this._authpacket, { force: true })
+            const promise =
+                this._optOutEventsArgs.length > 0
+                    ? this.call('optOutEvents', this._optOutEventsArgs, { force: true })
+                    : Promise.resolve();
+            promise
+                .then(() => this.emit('optOutResult', undefined))
```

Comment:
should be able to just skip passing in "undefined" here

Reply:
Yes this is in case you want to pass in the result from above promise, but I can change it.